### PR TITLE
test(blackbox): verify packaged direct recovery

### PR DESCRIPTION
**Pull request by lupuszr**
## What changed

PR #5 is rebased onto current `main`, which now contains PR #4's direct graph recovery. The ontology-prefetch production change is therefore intentionally gone: Blackbox requests only the configured public VM graph and DKG verifies its cleartext ID against the chain-committed name hash.

This PR now adds the missing regression proof requested in review:

- replace the synthetic `ok: true` plus `durableError` mock with the real DKG client failure shape: HTTP 503 `DURABLE_CATCHUP_ALL_PEERS_FAILED`
- add a package-level integration test that imports the built DKG JavaScript rather than DKG source files
- start that test with empty local subscription/binding maps
- prove a matching on-chain name hash authenticates and durably materializes the assertion
- prove a mismatched committed name hash fails closed with `VM_CHAIN_CONTEXT_GRAPH_MISMATCH`

## Release and safety gate

Current `main` requires DKG `10.0.9+` in both Unix and Windows installers. The currently published npm package is still `10.0.8`, so installation remains fail-closed until a release containing OriginTrail/dkg#1885 is published.

The integration test also checks for the exact packaged cold-join capability before running. It was validated against the `build-outputs` artifact produced by merged OriginTrail/dkg#1885.

## Resulting flow

```mermaid
sequenceDiagram
    participant BB as Blackbox
    participant DKG as Local DKG node
    participant Publisher as Curator publisher
    participant Chain as Blockchain
    participant Store as Local VM store

    BB->>DKG: Catch up configured public graph
    DKG->>Publisher: Fetch graph-scoped durable assertion
    DKG->>Chain: Read KA graph slot and committed graph-name hash
    alt Cleartext graph ID matches commitment
        DKG->>Store: Atomically materialize verified assertion
        DKG-->>BB: Verified transfer result
    else Missing or mismatched commitment
        DKG-->>BB: HTTP 503 all-peers-failed verification error
        BB-->>BB: Record failed synchronization
    end
```

No ontology graph is fetched in this flow, so the former five-pass ontology delay is eliminated rather than merely shortened.

## Validation

- focused direct-recovery tests: **2 passed**
- packaged DKG #1885 integration: **1 passed**
- Blackbox suite: **387 passed, 5 skipped, 1 known baseline deselected**
- Ruff: passed
- `git diff --check`: passed

The deselected baseline is the unchanged macOS executable-path assertion in `test_managed_dkg_sync_environment_changes_only_durable_mode`: the process reports the Python app-bundle path while the test expects the venv symlink path.

## Review focus

- Does the package test exercise the exact safety contract expected from the first DKG release containing #1885?
- Is keeping the installer fail-closed at `10.0.9+` acceptable until that package is published?

### CODE REVIEW

File: tests/plugins/test_blackbox_plugin.py [hunk 1]
```diff
@@ -1204,14 +1219,59 @@ def context_graphs(self):
...
+def test_authoritative_recovery_leaves_public_transfer_decision_to_dkg(monkeypatch):
+    graph_calls = []
+    states = []
+
+    class FakeClient:
+        def catchup_from_peer(self, cg_id, peer_id, *, budget_ms):
+            graph_calls.append(cg_id)
+            peer_result = {"peerId": peer_id}
+            if cg_id == constants.DEFAULT_CONTEXT_GRAPH_ID:
+                peer_result["durableError"] = "public VM verification rejected"
```
Comment (branarakic) [inline tests/plugins/test_blackbox_plugin.py]: This mock does not validate the dependency behavior on which the safety argument rests. The current DKG route reports a durable-only all-peer failure as HTTP 503, so DkgClient raises before returning this synthetic ok:true plus durableError shape; more importantly, no test proves that a fresh node with no ontology/registry binding accepts and stores the configured graph through chain name-hash verification. This resolution-chain and security-boundary change needs an integration test against the packaged DKG version, starting from an empty binding and asserting both durable insertion and rejection of a mismatched chain name hash.

File: plugins/blackbox/cli.py [hunk 2]
```diff
@@ -1564,39 +1565,37 @@ def _recover() -> None:
...
-                        "verified graph metadata did not expose a numeric "
-                        f"context-graph binding after {metadata_passes} passes"
-                    )
-                    sync_state.write(
-                        "failed",
-                        context_graph_id=context_graph_id,
-                        graph_peer_id=graph_peer_id,
-                        phase="resolving-context-graph-binding",
-                        error=error,
+                if metadata_passes < _MAX_GRAPH_METADATA_PASSES:
```
Comment (branarakic) [inline plugins/blackbox/cli.py]: This can still spend five serial passes on the lane whose result is now declared optional. Since public_progress_seen remains false during ontology recovery, each call gets INITIAL_GRAPH_SYNC_PASS_BUDGET_MS (30 seconds); the new test deliberately drives all five, so an unresolvable binding may consume about 150 seconds before the actual public graph is attempted and can exhaust a shorter overall deadline. Once the DKG chain-name-hash verifier is a prerequisite, please limit this to one genuinely best-effort prefetch or try the configured graph first.

File: plugins/blackbox/cli.py [hunk 3]
```diff
@@ -1365,10 +1365,11 @@ def _catchup_authoritative_vm(
...
 
-    # A fresh DKG edge does not yet know the cleartext graph id's numeric
-    # on-chain binding. The small system ontology graph carries that standard
-    # DKG metadata. Fetch it from the same source first so the VM verifier can
-    # authenticate each graph-scoped assertion as soon as it arrives.
+    # A fresh DKG edge may not yet know the cleartext graph id's numeric
+    # on-chain binding. Best-effort prefetch the small system ontology graph so
+    # the VM verifier can authenticate graph-scoped assertions immediately.
+    # The binding is only an optimization here: the DKG verifier, not this
+    # derived metadata query, remains authoritative for the public transfer.
```
Comment (branarakic) [inline plugins/blackbox/cli.py]: Blocking: this statement is only true with OriginTrail/dkg#1885. The currently published @origintrail-official/dkg latest is still 10.0.8, while both installers accept 10.0.7+. In 10.0.8 the durable verifier still resolves the local CG id through ontology; when that binding is missing it rejects the asset, so this branch merely advances to a public call that still fails for the affected installation. Please either require/install the first released package containing #1885 in both installers, or capability/version-gate this path and retain the explicit failure on older nodes.

**Review by branarakic — CHANGES_REQUESTED** [review]
Requesting changes on the release dependency and verification gate. The approach is safe with DKG #1885, but the currently published package is still 10.0.8 and the repo accepts older nodes where the missing ontology binding remains a hard verifier dependency. Please address the three inline findings before merge.
